src/In.ProjectEKA.HipService/Link/LinkPatient.cs

```diff
@@ -117,38 +108,33 @@ await discoveryRequestRepository.Delete(request.TransactionId, request.Patient.C
                                 patient.CareContexts.First(info => info.ReferenceNumber == context.ReferenceNumber)
                                     .Display)).ToList();
                         if (programs.Count != request.Patient.CareContexts.Count())
-                        {
                             return (null, new ErrorRepresentation(new Error(ErrorCode.CareContextNotFound,
                                 ErrorMessage.CareContextNotFound)));
-                        }
 
                         return (patient, (ErrorRepresentation) null);
                     })
                 .ValueOr((null,
                     new ErrorRepresentation(new Error(ErrorCode.NoPatientFound, ErrorMessage.NoPatientFound))));
         }
 
-        public virtual async Task<ValueTuple<PatientLinkConfirmationRepresentation, ErrorRepresentation>>
+        public virtual async Task<ValueTuple<PatientLinkConfirmationRepresentation, string, ErrorRepresentation>>
             VerifyAndLinkCareContext(
-                LinkConfirmationRequest request)
+            LinkConfirmationRequest request)
         {
-            var errorResponse = await patientVerification.Verify(request.LinkReferenceNumber, request.Token);
-            if (errorResponse != null)
-            {
-                return (null, new ErrorRepresentation(errorResponse.toError()));
-            }
-
             var (linkEnquires, exception) =
                 await linkPatientRepository.GetPatientFor(request.LinkReferenceNumber);
-
+            var cmId = "";
             if (exception != null)
-            {
-                return (null,
+                return (null,cmId,
                     new ErrorRepresentation(new Error(ErrorCode.NoLinkRequestFound, ErrorMessage.NoLinkRequestFound)));
-            }
+            cmId = linkEnquires.ConsentManagerId;
+
+            var errorResponse = await patientVerification.Verify(request.LinkReferenceNumber, request.Token);
+            if (errorResponse != null)
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6377136)